buildSrc/build.gradle.kts

```diff
@@ -54,29 +55,25 @@ subprojects {
 
         val compileGroovy: GroovyCompile by tasks
 
-        configurations {
-            "apiElements" {
-                outgoing.variants["classes"].artifact(mapOf(
-                    "file" to compileGroovy.destinationDir,
-                    "type" to ArtifactTypeDefinition.JVM_CLASS_DIRECTORY,
-                    "builtBy" to compileGroovy
-                ))
-            }
+        configurations.apiElements.apply {
+            outgoing.variants["classes"].artifact(mapOf(
+                "file" to compileGroovy.destinationDir,
+                "type" to ArtifactTypeDefinition.JVM_CLASS_DIRECTORY,
+                "builtBy" to compileGroovy
+            ))
         }
     }
     if (file("src/main/kotlin").isDirectory || file("src/test/kotlin").isDirectory) {
-        apply { plugin("kotlin") }
+        apply(plugin = "org.gradle.kotlin.kotlin-dsl")
 
         tasks.withType<KotlinCompile> {
             kotlinOptions {
                 freeCompilerArgs = listOf("-Xjsr305=strict")
             }
         }
     }
-    apply {
-        plugin("idea")
-        plugin("eclipse")
-    }
+    apply(plugin = "idea")
```

Comment:
Why do we prefer this syntax, where we have to repeat the context, in this case apply?

Reply:
`apply {}` is ambiguous due to the `kotlin-stdlib` extension function with the same name on `Any` that shadows our `apply {}`. This PR contains mechanical changes of *all* `apply {}` usages, favoring the unambiguous function. See the discussion in https://github.com/gradle/kotlin-dsl/issues/682 for more details.